java.base/src/main/java/java/lang/Class$_patch.java

```diff
@@ -44,6 +75,13 @@
     @NoReflect
     final ClassLoader classLoader;
 
+    /**
+     * ReflectionData for subset of members that have been annotated as available for runtime reflection
+     */
```

Comment:
One comment.  I tried to to a @Replace on the reflectionData field and for some reason it didn't work...the reflectionData() method ended up being compiled to a FieldNotFound exception.   I'd like to get to the bottom of this at some point, but decided to postpone dealing with it for now..

Reply:
Some of the field-matching code matches on just the name, and some matches on the name *and* descriptor. I bet you're hitting the latter. The spec says "If C declares a field with the name and descriptor specified by the field reference, field lookup succeeds. The declared field is the result of the field lookup."; I took "name and descriptor" to mean "name and type" at one point but I think the spec means "name and *owner's* descriptor", thus matching by exact type is wrong and should be fixed, at some point.